app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -143,6 +148,31 @@ class HomeFragmentPresenter @Inject constructor(
     return appHistoryResult.getOrDefault(UserAppHistory.getDefaultInstance())
   }
 
+  private val ongoingStoryListSummaryResultLiveData: LiveData<AsyncResult<OngoingStoryList>> by lazy {
+    topicListController.getOngoingStoryList()
+  }
+
+  private fun subscribeToOngoingStoryList() {
+    getAssumedSuccessfulOngoingStoryList().observe(fragment, Observer<OngoingStoryList> {
+      if (it.recentStoryCount > 0) {
+        it.recentStoryList.forEachIndexed { index, promotedStory ->
+          if (index < 3) {
```

Comment:
To make this simpler, I suggest using Kotlin's [take](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/take.html) function:

it.recentStoryList.take(3).forEach { ... }

I think this also lets you get rid of the recentStoryCount check (since retrieving an absent list from a proto defaults to an empty list).

Reply:
Done. Implemented this code.